web_src/js/features/codeeditor.js

```diff
@@ -164,13 +164,13 @@ export async function createCodeEditor(textarea, filenameInput, previewFileModes
 function getEditorConfigOptions(ec) {
   if (!isObject(ec)) return {};
 
-  const opts = {};
-  opts.detectIndentation = !('indent_style' in ec) || !('indent_size' in ec);
-  if ('indent_size' in ec) opts.indentSize = Number(ec.indent_size);
-  if ('tab_width' in ec) opts.tabSize = Number(ec.tab_width) || opts.indentSize;
-  if ('max_line_length' in ec) opts.rulers = [Number(ec.max_line_length)];
-  opts.trimAutoWhitespace = ec.trim_trailing_whitespace === true;
-  opts.insertSpaces = ec.indent_style === 'space';
-  opts.useTabStops = ec.indent_style === 'tab';
-  return opts;
+  return {
+    detectIndentation: !('indent_style' in ec) || !('indent_size' in ec),
+    trimAutoWhitespace: ec.trim_trailing_whitespace === true,
+    insertSpaces: ec.indent_style === 'space',
+    useTabStops: ec.indent_style === 'tab',
+    ...('indent_size' in ec && {indentSize: Number(ec.indent_size)}),
```

Comment:
Is there something different than using `identSize: 'indent_size' in ec ? Number(ec.indent_size) : null`? Seems overkill to me to spread objects here for optional arguments.

Reply:
depends on how monaco handles `null` properties. I wanted to avoid such questions by not specifying the argument at all :)

Reply:
I don't find this particular good taste of Javascript(the previous implementation is better in this regard IMO). 



> depends on how monaco handles `null` properties.

I hope that they have a small sanity check before using the options.